lib/std/crypto/bcrypt.zig

```diff
@@ -764,20 +769,25 @@ pub const HashOptions = struct {
 /// IMPORTANT: by design, bcrypt silently truncates passwords to 72 bytes.
 /// If this is an issue for your application, set the `silently_truncate_password` option to `false`.
 pub fn strHash(
+    allocator: mem.Allocator,
     password: []const u8,
-    options: HashOptions,
-    out: []u8,
-) Error![]const u8 {
-    switch (options.encoding) {
-        .phc => return PhcFormatHasher.create(password, options.params, out),
-        .crypt => return CryptFormatHasher.create(password, options.params, out),
-    }
+    comptime options: HashOptions,
+) Error![]u8 {
+    const buf_len = comptime if (options.strhash_max_bytes) |len| len else switch (options.encoding) {
+        .crypt => hash_length,
+        .phc => hash_length * 2,
+    };
+    var buf: [buf_len]u8 = undefined;
```

Comment:
maybe provide `strHash` and `strHashAlloc` for bcrypt? We still use the stack based approach in std, but don't give the developer any choice if they know what they're doing